v2/client/http_receiver.go

```diff
@@ -25,8 +26,11 @@ type EventReceiver struct {
 }
 
 func (r *EventReceiver) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
 	go func() {
 		r.p.ServeHTTP(rw, req)
+		wg.Done()
```

Comment:
I admittedly don't have the full context just saw this floating by so apologies for likely noise :) 
Can't we just call this synchronously instead of in a func? Am I missing something? Perhaps worthy of a comment so it's easier to reason about, but it's probably just me :)

Reply:
There are internal details of the SDK that require these two parts to work in parallel based on how the bindings work is documented, so yeah the server needs to go and it will block until the receive call is finished which is the second block of blocking call, but the bug was the function can't exit until ServeHTTP has finished, which is blocked on the callback... 

yikes